Revert "do not merge: always force parquet format in cdc tests"

jayshrivastava opened the pull request:
### Revert "do not merge: always force parquet format in cdc tests" #103928

This reverts commit 63eccd4a513dfea440a2f7d6176d759e4c8a9eba.

Release note: None
Epic: None

---

### changefeedccl: refactor parquet column iterators

Previously, some tests would fail due to invalid conversion
from parquet to JSON in testing. These failures showed two
underlying issues with how columns were written:
- in tests watching a column family not containing the primary
  key of the table, we would not write the primary key to
  the parquet file
- in tests where the primary key was defined with a particular
  order which is different than the table, the parquet testing
  code would not order the keys correctly

This change fixes the two above issues by (a) using the correct
iterators on `cdcevent.Row` and writing more verbose
metadata to the parquet file to use in testing.

Epic: None
Release note: None

jayshrivastava committed:
Revert "do not merge: always force parquet format in cdc tests if possible"

This reverts commit 63eccd4a513dfea440a2f7d6176d759e4c8a9eba.

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103928)
<!-- Reviewable:end -->

jayshrivastava committed:
changefeedccl: refactor parquet column iterators

Previously, some tests would fail due to invalid conversion
from parquet to JSON in testing. These failures showed two
underlying issues with how columns were written:
- in tests watching a column family not containing the primary
  key of the table, we would not write the primary key to
  the parquet file
- in tests where the primary key was defined with a particular
  order which is different than the table, the parquet testing
  code would not order the keys correctly

This change fixes the two above issues by (a) using the correct
iterators on `cdcevent.Row` and writing more verbose
metadata to the parquet file to use in testing.

Epic: None
Release note: None

jayshrivastava commented:
bors r=HonoreDB
